### ISSUE TITLE
Renewed maintenance of qt5.cr

### ISSUE THREAD

**Issue by docelic** [Papierkorb/qt5.cr#29]
Hello @acoolstraw, @dscottboggs , @kalinon , @glenux , @ZaWertun, @gilmartaj, @codenoid, @foliea

Same as for bindgen (https://github.com/Papierkorb/bindgen/issues/23), I would like to work on qt5.cr so that it builds/works again.

Any interest or contributions from you would be much appreciated - either in the form of PRs or general comments/insights (possibly added to this thread or whatever is more appropriate).

Let's try to build a critical mass here so that we succeed in updating the Qt bindings.

(For example, reviewing your existing PRs or updating your forks to the latest version of bindgen so that it is easier to identify your own changes/improvements would be a great start, and we could coordinate from there.)

Also let's please keep any patches/PRs small and isolated so that they are easier to review and agree on.

Thanks!

**Comment by shayneoneill** [comment]
Amazing, awesome and I'm totally down with cheering this on. So much potential for being the "Killer app" of Crystallang

**Comment by shayneoneill** [comment]
Yeah still a bit mystified on how to get this to work with MacOS. I'm taking the 'just works' claim as aspirational rather than current. I've downgraded LLVM back to v7 , full install view homebrew, still the same deal, it cant find clang. Unfortuantely I'm just finding no guidance anywhere on how to proceed with this.

**Comment by nogweii** [comment]
I'm excited to see qt5.cr restarted! I was looking for a language to start writing some applications in, and have been curious about using Crystal for "real" work for a while. I look forward to trying it out soon!

**Comment by docelic** [comment]
@shayneoneill yes, I intend to test on MacOS and Windows Subsystem for Linux as well.

**Comment by shayneoneill** [comment]
Spectacular. Looking forward to it.

**Comment by docelic** [comment]
I was able to generate bindings for Qt 5.12 and run the sample programs included with qt5.cr by using system's Qt 5.12 libraries.

**Comment by ShalokShalom** [comment]
I was witnessing the birth of this project, and happen to be a huge qt fan. Papierkorb demonstrated extraordinarily quick and competent response and I was looking forward to a qt-quick integration, so that I can start working on apps.

Looking through the available list of qt implementations, Crystal quickly turns out as one of the only modern languages that has proper support for it. So I am very horny to give it a try, while qt widget becomes more and more mainly used to maintain already existing desktop applications, while qt quick and qml show great potential to be one of the go-to solutions to write software that runs stationary and mobile. 

And a lot of people think very similar.

**Comment by shayneoneill** [comment]
Still havent been able to progress on getting things to build 

```
shayneoneill@Shaynes-New-2015 qt2 % qtdiag 
Qt 5.15.1 (x86_64-little_endian-lp64 shared (dynamic) release build; by Clang 11.0.3 (clang-1103.0.32.62) (Apple)) on "cocoa" 
OS: macOS 10.16 [darwin version 20.1.0]

Architecture: x86_64; features: SSE2 SSE3 SSSE3 SSE4.1 SSE4.2 AVX AVX2

```

and

```
shayneoneill@Shaynes-New-2015 qt2 % crystal run test.cr 
Showing last frame. Use --error-trace for full trace.

In lib/qt5/src/qt5/binding.cr:7:20

 7 | {% use_binding = run("#{__DIR__}/../../support/decide_binding_fast.cr") %}
                      ^--
Error: Error executing run (exit code: 1): /Users/shayneoneill/crystal/qt2/lib/qt5/src/qt5/../../support/decide_binding_fast.cr 
```
On the Crytal 0.35.1 / MacOS Big Sur

**Comment by codenoid** [comment]
Finally crystal hit 1.0, how this going

**Comment by docelic** [comment]
Hey @codenoid , it seems 1.0.0 is not exactly released yet. 
But many things were added to bindgen in the meantime.
We're discussing a new release of bindgen and of Qt bindings here: https://github.com/Papierkorb/bindgen/issues/107
